packages/calypso-e2e/src/lib/pages/pages-page.ts

```diff
@@ -19,20 +19,25 @@ export class PagesPage {
 	/**
 	 * Opens the Pages page.
 	 *
+	 * @param param0 Keyed object parameter.
+	 * @param {string} param0.siteSlug Site slug.
+	 * @param {number} param0.timeout Custom timeout.
+	 *
 	 * Example {@link https://wordpress.com/pages}
+	 * Example {@link https://wordpress.com/pages/usersiteslug.wordpress.com}
 	 */
-	async visit(): Promise< Response | null > {
-		const response = await this.page.goto( getCalypsoURL( 'pages' ) );
-		return response;
+	async visit( { siteSlug, timeout }: { siteSlug?: string; timeout?: number } = {} ) {
+		const url = `pages/${ siteSlug }`;
+		await this.page.goto( getCalypsoURL( url ), { timeout: timeout } );
+
+		// Wait for page entries (if any) to load. This also waits for the page to settle.
+		await this.page.locator( '.is-placeholder' ).waitFor( { state: 'detached', timeout: timeout } );
```

Comment:
Intended to make visiting the the `/pages` endpoint more stable. Waits for the placeholder elements to disappear.